pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -29,5 +33,26 @@ func New(cfg *setting.Cfg) httpclient.Provider {
 
 	return newProviderFunc(sdkhttpclient.ProviderOptions{
 		Middlewares: middlewares,
+		ConfigureTransport: func(opts sdkhttpclient.Options, transport *http.Transport) {
+			datasourceName, exists := opts.Labels["datasource_name"]
+			if !exists {
+				return
+			}
+			newConntrackRoundTripper(datasourceName, transport)
+		},
 	})
 }
+
+// newConntrackRoundTripper takes a http.DefaultTransport and adds the Conntrack Dialer
+// so we can instrument outbound connections
+func newConntrackRoundTripper(name string, transport *http.Transport) *http.Transport {
+	transport.DialContext = conntrack.NewDialContextFunc(
+		conntrack.DialWithTracing(),
```

Comment:
Let's skip this one for now. Still have no idea how to expose this data. Perhaps something to look into later together with adding support for conntrack to grafana server?